File: Bit_Manipulation/python/conversion.py
from numpy import uint32


def get_bit(num, i):
    return ((num & (1 << i)) != 0)


def get_ones_count(num):
    count = 0

    for i in range(32):
        if get_bit(num, i):
            count += 1

    return count


def determine_number(num1, num2):
    return get_ones_count(num1 | num2) - get_ones_count(num1 & num2)


def main():
    num1 = uint32(244)
    num2 = uint32(249)

    print("{0:b}".format(num1))
    print("{0:b}".format(num2))

    print(determine_number(num1, num2))


if __name__ == '__main__':
    main()

File: Bit_Manipulation/go/conversion.go
package main

import (
  "fmt"
)


func getBit(num uint32, i int) bool {
  return ((num & (1 << i)) != 0)
}


func getOnesCount(num uint32) int {
  var count int

  for i := 0; i < 32; i++ {
    if getBit(num, i) {
      count++
    }
  }

  return count
}


func determineNumber(num1 uint32, num2 uint32) int {
  return getOnesCount(num1 | num2) - getOnesCount(num1 & num2)
}


func main() {
  var a, b uint32
  a = 342
  b = 330

  fmt.Printf("%032b\n", a)
  fmt.Printf("%032b\n\n", b)

  fmt.Println(determineNumber(a, b))
}
